app/src/main/java/com/example/tasky/common/presentation/editing/Editing.kt

```diff
@@ -0,0 +1,207 @@
+        placeholderText = getPlaceholderText(textFieldType, agendaDetailsType)
+    )
+}
+
+@Composable
+fun EditScreenContent(
+    toolbarTitle: String,
+    navigateToPreviousScreen: () -> Unit,
+    saveText: (String) -> Unit,
+    text: String,
+    placeholderText: String,
+    onTextChange: (String) -> Unit
+) {
+    Box(
+        modifier = Modifier
+            .fillMaxSize()
+            .background(Color.White)
+            .safeDrawingPadding()
+    ) {
+        Column {
+            EditToolbar(
+                toolbarTitle = toolbarTitle,
+                navigateToPreviousScreen = navigateToPreviousScreen,
+                saveEventTitle = { saveText(text) }
+            )
+            GrayDivider()
+            EditableFieldArea(
+                text = text,
+                placeholderText = placeholderText,
+                onTextChange = onTextChange
+            )
+        }
+    }
+}
+
+@Composable
+fun EditableFieldArea(
+    text: String,
+    placeholderText: String,
+    onTextChange: (String) -> Unit
+) {
+    TextField(
+        value = text,
+        onValueChange = onTextChange,
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(horizontal = 16.dp),
+        singleLine = false,
+        placeholder = { Text(placeholderText, fontSize = 26.sp) },
+        textStyle = TextStyle(fontSize = 26.sp),
+        colors = TextFieldDefaults.colors(
+            focusedContainerColor = Color.White,
+            unfocusedContainerColor = Color.White,
+            disabledContainerColor = Color.White,
+            focusedIndicatorColor = Color.Transparent,
+            unfocusedIndicatorColor = Color.Transparent,
+            disabledIndicatorColor = Color.Transparent,
+        )
+    )
+}
+
+@Composable
+fun EditToolbar(
+    toolbarTitle: String,
+    navigateToPreviousScreen: () -> Unit,
+    saveEventTitle: () -> Unit
+) {
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(start = 16.dp, end = 16.dp, top = 20.dp, bottom = 5.dp),
+        horizontalArrangement = Arrangement.SpaceBetween,
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        LeftCarrotIcon(
+            modifier = Modifier.clickable { navigateToPreviousScreen() }
+        )
+        EditTitle(toolbarTitle = toolbarTitle)
+        SaveButton(onClick = { saveEventTitle() })
+    }
+}
+
+@Composable
+fun EditTitle(toolbarTitle: String) {
+    Text(
+        text = toolbarTitle,
+        color = Color.Black,
+        fontWeight = FontWeight.SemiBold,
+        fontSize = 18.sp
+    )
+}
+
+@Composable
+fun SaveButton(onClick: () -> Unit) {
+    Text(
+        text = stringResource(R.string.save),
+        color = colorResource(id = R.color.tasky_green),
+        fontWeight = FontWeight.SemiBold,
+        fontSize = 16.sp,
+        modifier = Modifier.clickable { onClick() }
+    )
+}
+
+@Composable
+fun getPlaceholderText(textFieldType: String, agendaDetailsType: String): String {
```

Comment:
Could it make sense to work with enums here instead to make this less error-prone?

Reply:
note: convert string back to enum after navigation